src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2197,10 +2195,10 @@ public override IOperation VisitConditionalAccess(IConditionalAccessOperation op
                 if (ITypeSymbolHelpers.IsNullableType(operation.Type) && !ITypeSymbolHelpers.IsNullableType(currentConditionalAccess.WhenNotNull.Type))
                 {
                     IOperation access = Visit(currentConditionalAccess.WhenNotNull);
-                    AddStatement(new FlowCapture(resultCaptureId, currentConditionalAccess.WhenNotNull.Syntax,
-                                                 TryMakeNullableValue((INamedTypeSymbol)operation.Type, access, compilation) ??
-                                                 // PROTOTYPE(dataflow): The scenario with missing constructor is not covered by unit-tests.
-                                                 MakeInvalidOperation(operation.Type, access)));
```

Comment:
I don't think there is a need to do any additional checks here, just replace the coalesce with ```MakeNullable(access, operation.Type)``` and revert the refactoring.